src/libstd/f32.rs

```diff
@@ -908,10 +908,9 @@ impl f32 {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn asinh(self) -> f32 {
-        if self == NEG_INFINITY {
-            NEG_INFINITY
-        } else {
-            (self + ((self * self) + 1.0).sqrt()).ln()
+        match self {
+            x if x == NEG_INFINITY => NEG_INFINITY,
+            x => (x + ((x * x) + 1.0).sqrt()).ln().copysign(self)
```

Comment:
I would prefer to use if-else directly.

Reply:
Yea, that sounds reasonable, maybe consider changing the original `acosh` BTW?
```rust
pub fn acosh(self) -> f64 {
        match self {
            x if x < 1.0 => NAN,
            x => (x + ((x * x) - 1.0).sqrt()).ln(),
        }
    }
```